eclair-core/src/main/scala/fr/acinq/eclair/channel/publish/ReplaceableTxFunder.scala

```diff
@@ -317,9 +317,9 @@ private class ReplaceableTxFunder(nodeParams: NodeParams,
   private def sign(fundedTx: ReplaceableTxWithWitnessData, txFeerate: FeeratePerKw, amountIn: Satoshi): Behavior[Command] = {
     val channelKeyPath = keyManager.keyPath(cmd.commitment.localParams, cmd.commitment.params.channelConfig)
     fundedTx match {
-      case claimAnchorTx: ClaimLocalAnchorWithWitnessData =>
-        val localSig = keyManager.sign(claimAnchorTx.txInfo, keyManager.fundingPublicKey(cmd.commitment.localParams.fundingKeyPath, cmd.commitment.fundingTxIndex), TxOwner.Local, cmd.commitment.params.commitmentFormat)
-        val signedTx = claimAnchorTx.copy(txInfo = addSigs(claimAnchorTx.txInfo, localSig))
+      case ClaimLocalAnchorWithWitnessData(anchorTx) =>
+        val localSig = keyManager.sign(anchorTx, keyManager.fundingPublicKey(cmd.commitment.localParams.fundingKeyPath, cmd.commitment.fundingTxIndex), TxOwner.Local, cmd.commitment.params.commitmentFormat)
+        val signedTx = ClaimLocalAnchorWithWitnessData(addSigs(anchorTx, localSig))
```

Comment:
My reasoning here is on the contrary, that it's safer to construct it entirely, because if you call `copy`, you cannot be sure that you didn't forget to also update another field in the object? Whereas recreating the object forces you to make sure all fields are correctly updated.

Reply:
I disagree: `.copy()` (which we use in our code base a lot) makes the code and its intent really clear: if a field should have been updated and was not it will be easy to spot, and that would be a bug. Creating a new objects could fail in ways that are not easy to spot (if there are multiple constructors with different parameters for example).

Reply:
> it will be easy to spot, and that would be a bug

I don't think it will be easy to spot at all?

Imagine for example that we add a field to `ClaimLocalAnchorWithWitnessData`. If we use `.copy`, we need to manually scan the whole codebase for every place where we copy instance of that field and verify that we update (or not) the new field. If we use a constructor, then the compiler automatically tells us and we have to decide for each object creation what values to put in that field.

If you're still not convinced by that argument I can change it, but I think it's a mistake...

Reply:
Either way seems fine to me, as long as our policy is to not introduce fields with default values to existing classes.

\start(bikeshedding)
I generally like `.copy` because it makes it explicit what the intent is, rather then needing to parse the line visually for what changed; manually copying fields might also introduce an error as mentioned if a new field is added with a default value. However, I believe our policy is (or should be) to avoid adding fields with default values to existing classes.  In this case, the compiler will warn us everywhere the class is initiated and trigger any analysis of places it should change independent of when `.copy` is used. 
\end(bikeshedding)

Reply:
It's true that the intent is clearer with `.copy`, anyway this part of the code is extensively tested so it should catch potential issues. I'll change it to `.copy`.

Reply:
Done in https://github.com/ACINQ/eclair/pull/2726/commits/9ff76f49b0a013329262e8b1ad6e10ef6aa88cfb